packages/flutter/lib/src/gestures/resampler.dart

```diff
@@ -87,7 +87,7 @@ class PointerEventResampler {
       device: event.device,
       position: position,
       delta: delta,
-      buttons: event.buttons,
+      buttons: event.buttons | kPrimaryButton,
```

Comment:
`event.buttons | kPrimaryButton `just make sure to have at least the default value kPrimaryButton.
The values ​​of kPrimaryButton, kSecondaryButton and kTertiaryButton are all greater than 0.
``` dart
const int kPrimaryButton = 0x01;
const int kSecondaryButton = 0x02;
const int kTertiaryButton = 0x04;
```

Reply:
> I don't think we should be changing this here as it would result in incorrect behavior if kSecondaryButton was pressed.
> 
> Can we instead introduce a _hasButtons field (`int _hasButtons = 0;` after line 43) that is updated from event.buttons just after line 211, add `final int hadButtons = _hasButtons;` just after line 207, and then pass the hadButtons value to _toMoveOrHoverEvent at line 238?
> 
> You'll have to update _toMoveOrHoverEvent, _toMoveEvent and _toHoverEvent to take a `int buttons` argument in addition to the existing arguments for this.

`event.buttons | kPrimaryButton `just make sure to have at least the default value kPrimaryButton.
The values ​​of kPrimaryButton, kSecondaryButton and kTertiaryButton are all greater than 0.
``` dart
const int kPrimaryButton = 0x01;
const int kSecondaryButton = 0x02;
const int kTertiaryButton = 0x04;
```

Reply:
> I don't think we should be changing this here as it would result in incorrect behavior if kSecondaryButton was pressed.
> 
> Can we instead introduce a _hasButtons field (`int _hasButtons = 0;` after line 43) that is updated from event.buttons just after line 211, add `final int hadButtons = _hasButtons;` just after line 207, and then pass the hadButtons value to _toMoveOrHoverEvent at line 238?
> 
> You'll have to update _toMoveOrHoverEvent, _toMoveEvent and _toHoverEvent to take a `int buttons` argument in addition to the existing arguments for this.

Hi, please help me to review again.